jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -887,17 +902,60 @@ public void setDeflaterPoolCapacity(int capacity)
         if (isStarted())
             throw new IllegalStateException(getState());
 
-        poolCapacity = capacity;
+        _deflaterPool.setCapacity(capacity);
     }
 
-    protected DeflaterPool newDeflaterPool(int capacity)
+    /**
+     * Gets the maximum number of Inflators that the DeflaterPool can hold.
+     *
+     * @return the Deflater pool capacity
+     */
+    public int getInflaterPoolCapacity()
     {
-        return new DeflaterPool(capacity, Deflater.DEFAULT_COMPRESSION, true);
+        return _inflaterPool.getCapacity();
+    }
+
+    /**
+     * Sets the maximum number of Inflators that the DeflaterPool can hold.
+     */
+    public void setInflaterPoolCapacity(int capacity)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+
+        _inflaterPool.setCapacity(capacity);
+    }
+
+    protected InflaterPool newInflaterPool()
+    {
+        return new InflaterPool(CompressionPool.INFINITE_CAPACITY, true);
+    }
+
+    protected DeflaterPool newDeflaterPool()
+    {
+        return new DeflaterPool(CompressionPool.INFINITE_CAPACITY, Deflater.DEFAULT_COMPRESSION, true);
     }
 
     @Override
     public String toString()
     {
         return String.format("%s@%x{%s,min=%s,inflate=%s}", getClass().getSimpleName(), hashCode(), getState(), _minGzipSize, _inflateBufferSize);
     }
+
+    public static class CaseInsensitiveSet extends HashSet<String>
```

Comment:
This can be simplified to just ...

``` java
    public static class CaseInsensitiveSet extends TreeSet<String>
    {
        public CaseInsensitiveSet()
        {
            super(String.CASE_INSENSITIVE_ORDER);
        }
    }
```

Reply:
I thought about this, won't this slow access down from O(1) to O(log n), and we don't really need it to be sorted. Not sure how much this would actually impact the performance.

Reply:
There's only what? like 5 DispatcherTypes?
And the only impact is on setup/configuration/initialization, no real impact during runtime.

Reply:
This is more MIME types, but yeah probably won't be that many in the set.
The set should be accessed on every request.

Reply:
Would a `AsciiLowerCaseSet` in jetty-util be useful (in general)?

Reply:
Another difference ... TreeSet doesn't allow null entries. HashSet does.

Reply:
I think it could be useful in jetty-util, although I don't know where else it could be used.

Reply:
I have now moved this to jetty-util.